test/unit_test/passes/pytorch/test_torch_trt_conversion.py

```diff
@@ -0,0 +1,85 @@
+# --------------------------------------------------------------------------
+import sys
+import tempfile
+from pathlib import Path
+from unittest.mock import MagicMock, patch
+
+import torch
+
+import olive.passes.pytorch.sparsegpt_utils as sparsegpt_utils
+from olive.data.template import huggingface_data_config_template
+from olive.model import PyTorchModel
+from olive.passes.olive_pass import create_pass_from_dict
+from olive.passes.pytorch import TorchTRTConversion
+from olive.systems.local import LocalSystem
+
+
+class MockTRTLinearLayer(torch.nn.Module):
+    pass
+
+
+def mocked_torch_zeros(*args, **kwargs):
+    if "device" in kwargs:
+        del kwargs["device"]
+    return torch.ones(*args, **kwargs) * 0
+
+
+# all the following patches are needed to run the test on CPU
+# make torch.cuda.is_available return True
+@patch("torch.cuda.is_available", return_value=True)
+# make tensor_data_to_device return the input tensor, ignore device
+@patch("olive.common.utils.tensor_data_to_device", side_effect=lambda x, y: x)
+# make torch.nn.Module.to do nothing
+@patch("torch.nn.Module.to", side_effect=lambda device: None)
+# replace device in kwargs with "cpu"
+@patch("torch.zeros", side_effect=mocked_torch_zeros)
+def test_sparsegpt(mock_is_cuda_available, mock_tensor_data_to_device, mock_module_to, mock_torch_zeros):
+    # setup
+    # mock trt utils since we don't have tensorrt and torch-tensorrt installed
+    mock_trt_utils = MagicMock()
+    mock_compile_trt_model = MagicMock()
+    mock_compile_trt_model.return_value = MockTRTLinearLayer()
+    mock_trt_utils.compile_trt_model = mock_compile_trt_model
+
+    sys.modules["olive.passes.pytorch.trt_utils"] = mock_trt_utils
+    with tempfile.TemporaryDirectory() as tempdir:
+        local_system = LocalSystem()
+        model_name = "hf-internal-testing/tiny-random-OPTForCausalLM"
+        task = "text-generation"
+        model_type = "opt"
+        input_model = PyTorchModel(hf_config={"model_name": model_name, "task": task})
+        original_submodules = list(
+            sparsegpt_utils.get_layer_submodules(
+                sparsegpt_utils.get_layers(input_model.load_model(), model_type)[0], submodule_types=[torch.nn.Linear]
+            ).keys()
+        )
+
+        dataset = {
+            "data_name": "ptb_text_only",
+            "subset": "penn_treebank",
+            "split": "train",
+            "input_cols": ["sentence"],
+            "component_kwargs": {"pre_process_data": {"seqlen": 100, "max_samples": 1, "random_seed": 42}},
+        }
+        data_config = huggingface_data_config_template(model_name=model_name, task=task, **dataset)
+        config = {
+            "data_config": data_config,
+        }
+
+        p = create_pass_from_dict(TorchTRTConversion, config, disable_search=True)
+        output_folder = str(Path(tempdir) / "sparse")
+
+        # execute
+        model = local_system.run_pass(p, input_model, None, output_folder)
```

Comment:
>local_system.run_pass [](http://example.com/codeflow?start=16&length=21)

could the pass.run be more clear?

Reply:
The other pass tests do the same as this so I copied them here

Reply:
Resolving this conversation since this is the case for all pass tests. Maybe we can make the change to all pass tests in a different PR if nedded. 
Please unresolve the conversation if you think otherwise.